Library/QuickPulseStateManager.ts

```diff
@@ -124,14 +125,15 @@ class QuickPulseStateManager {
         }
 
         let currentTimeout = QuickPulseStateManager._isCollectingData ? 1000 : 5000;
-        if (QuickPulseStateManager._isCollectingData && Date.now() - this._lastSuccessTime >= 20000) {
+        if (QuickPulseStateManager._isCollectingData && Date.now() - this._lastSuccessTime >= 20000 && !this._lastSendSucceeded) {
             // Haven't posted successfully in 20 seconds, so wait 60 seconds and ping
             QuickPulseStateManager._isCollectingData = false;
             currentTimeout = 60000;
-        } else if (!QuickPulseStateManager._isCollectingData && Date.now() - this._lastSuccessTime >= 60000) {
+        } else if (!QuickPulseStateManager._isCollectingData && Date.now() - this._lastSuccessTime >= 60000 && !this._lastSendSucceeded) {
             // Haven't pinged successfully in 60 seconds, so wait another 60 seconds
             currentTimeout = 60000;
         }
+        this._lastSendSucceeded = null;
```

Comment:
>null [](start = 34, length = 4)

Why not set this to false?

Reply:
Meant to convey a "pending" state versus having explicitly failed ("false")